### PR TITLE
Migrate to pnpm and changesets

### PR DESCRIPTION
**Pull request by jakebailey**
This switches the repo to pnpm/changesets, the monorepo and deployment method we use for dt-tools and attw (and me personally).

Notably:

- Internal deps are now marked as `workspace:*`, so we don't need to do the bumping manually.
- Publishing is now via `pnpm publish -r`, which rewrites the specifiers automatically.
- Changesets auto-bumps the version numbers themselves.
- Builds are a lot faster (multiple minutes better).

TODO:

- [x] Figure out the changesets private package setting; see also https://github.com/microsoft/DefinitelyTyped-tools/pull/1003
- [x] Perhaps it should be `workspace:^` to be less strict.
	- For now, no, as we used explicit deps previously.
- [x] Docs on how to use changesets
- [x] Make CI expectations happy

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "fixed": [],
+  "linked": [],
+  "access": "public",
+  "baseBranch": "v2",
+  "updateInternalDependencies": "patch",
+  "ignore": [],
+  "privatePackages": {
+    "tag": false,
+    "version": false
+  }
```
Comment (jakebailey) [inline .changeset/config.json:14]: I didn't know about these but they prevent changeset from caring about private packages.

**Comment by jakebailey** [comment]
Hey, it works.

File: .changeset/config.json (hunk 1 above)
Comment (jakebailey) [reply .changeset/config.json:14]: Doesn't seem to work: https://github.com/microsoft/DefinitelyTyped-tools/pull/1003

**Comment by jakebailey** [comment]
I enabled testing on Windows which has exposed that the VFS doesn't pass. May need to re-disable tests there.

**Comment by jakebailey** [comment]
Yay, got it working on all OSs.

**Review by andrewbranch — APPROVED** [review]
Amazing. What’s the remaining CI status thing?

**Comment by jakebailey** [comment]
> Amazing. What’s the remaining CI status thing?

GitHub is configured to require the CI check named "tests" but that doesn't work when it turns into a matrix as the name changed. Which is silly. I plan to use this trick: https://github.com/microsoft/typescript-benchmarking/blob/7371a75e3d5862eaffabaca749b94cfa9b934bc9/.github/workflows/ci.yml#L132

**Comment by jakebailey** [comment]
This is looking good to go, but I'm going to do one last check and pack each package and compare it to what's on npm to ensure that the build is actually working as intended.

**Review by navya9singh — APPROVED** [review]

**Comment by jakebailey** [comment]
Diffing was fruitful and found a few issues. The packages that can be published after this PR should be correct, plus or minus that the other gatsby 5 and such PR updated some deps.

**Pull request merged**